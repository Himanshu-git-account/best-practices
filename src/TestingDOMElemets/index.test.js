import React from 'react'
import {render,fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import TestDOMElements from '.'

describe('Testing DOM Elements',()=>{
    describe('When Component renders',()=>{
        it('should display 0 as initial number',()=>{
            const {getByTestId} = render(<TestDOMElements />)
            expect(getByTestId('number')).toHaveTextContent(0)
        })
        it('should increment the number when Increment button is clicked',()=>{
            const {getByTestId} = render(<TestDOMElements />)
            fireEvent.click(getByTestId('increment-button'))
            expect(getByTestId('number')).toHaveTextContent(1)
        })
        it('should take a snapshot', () => {
            const { asFragment } = render(<TestDOMElements />)
            
            expect(asFragment(<TestDOMElements />)).toMatchSnapshot()
        })
        
    })

   
})