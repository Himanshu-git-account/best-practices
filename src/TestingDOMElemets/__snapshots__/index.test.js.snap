// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DOM Elements When Component renders should take a snapshot 1`] = `
<DocumentFragment>
  <h1
    data-testid="number"
  >
    0
  </h1>
  <button
    data-testid="increment-button"
  >
     Increment
  </button>
  <button
    data-testid="decrement-button"
  >
    Decrement
  </button>
</DocumentFragment>
`;
